File: C/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int L;
int W;
int S;

void letters(string s);
float average(int arg1);

int main(void)
{
    // while(true){
    string input = get_string("Input your text here : ");
    letters(input);

    // printf("%i %i %i\n",L,W,S);
    //                    29 / 3 * 100  56.84          4 / 8 * 100 14.8
    float index = (0.0588 * average(L)) - (0.296 * average(S)) - 15.8;
    int tmp = (int) index;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if ((int)(index + 0.5) > tmp)
    {
        tmp = (int) index + 1;
        // if (tmp > 16)
        printf("Grade %i\n", tmp);
    }
    else if (tmp > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", tmp);
    }
    // printf("%i\n%i\n%i\n",L,W,S);
    // }
}

void letters(string s)
{
    // if a-z,
    for (int i = 0, l = strlen(s); i < l; i++)
    {
        if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            L++;
        }
        if (s[i] == ' ')
        {
            W++;
        }
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            S++;
        }
    }
    W++;
}

float average(int arg1)
{
    return ((float) arg1 * 100 / (float) W);
}

File: python/sentimental-readability/readability.py
import cs50


def main():
    input_string = (input("Input your text here : "))

    sentence_counter = 0
    word_counter = 0
    letter_counter = 0

    for words in input_string:
        if words == '.' or words == '?' or words == '!':
            sentence_counter += 1

    strings_words = []
    strings_words = input_string.split()
    word_counter = len(strings_words)

    # print(input_string)
    # print(strings_words)
    # print(word_counter)

    letters_list = []
    for letters in input_string:
        if not (letters == '.' or letters == ',' or
                letters == '?' or letters == ' ' or
                letters == '!' or letters == ':' or
                letters == ';' or letters == '\''):
            letters_list.append(letters.lower())

    letter_counter = len(letters_list)

    # print(letters_list)
    print(letter_counter)
    print(word_counter)
    print(sentence_counter)

    index = (0.0588 * average(letter_counter, word_counter)) - (0.296 * average(sentence_counter, word_counter)) - 15.8
    index = round(index)

    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def letters(input):
    for i in len(input):
        print(i)


def average(a, b):
    return a * 100 / b


if __name__ == "__main__":
    main()